### PR TITLE
Fixed: operations on reminders should not be allowed if the sidecar isn't hosting actors of that type

### LINKED ISSUE #6150: Actor reminder replacement silently fails if the sidecar doesn't host actors of the given type
## In what area(s)?

/area runtime

## What version of Dapr?

> master
> v1.10.x

## Description

Start Dapr without an app:

```sh
cd cmd/daprd
go run . \
  --app-id testreminders \
  --resources-path resources \
  --placement-host-address localhost:50005 \
  --dapr-http-port 3500
```

(You will need to have a a Component YAML in the `resources` folder that is configured as actorStateStore).

On another terminal:

```sh
curl -l --request POST 'http://localhost:3500/v1.0/actors/myactor/actor1/reminders/testreminder3' \
  --header 'Content-Type: application/json' \
  --data-raw '{"dueTime": "2h","period": "1h"}'
curl 'http://localhost:3500/v1.0/actors/myactor/actor1/reminders/testreminder3'

# Output: {"period":"1h","dueTime":"2h"}
```

Then **stop and restart** the sidecar in the first terminal.

Then run in the second terminal:

```sh
curl -l --request POST 'http://localhost:3500/v1.0/actors/myactor/actor1/reminders/testreminder3' \
  --header 'Content-Type: application/json' \
  --data-raw '{"dueTime": "14h","period": "2h"}'
curl 'http://localhost:3500/v1.0/actors/myactor/actor1/reminders/testreminder3'
# Output is wrong: {"period":"1h","dueTime":"2h"}
```

Repeat:

```sh
curl -l --request POST 'http://localhost:3500/v1.0/actors/myactor/actor1/reminders/testreminder3' \
  --header 'Content-Type: application/json' \
  --data-raw '{"dueTime": "14h","period": "2h"}'
curl 'http://localhost:3500/v1.0/actors/myactor/actor1/reminders/testreminder3'
# Output is now right: {"period":"2h","dueTime":"14h"}
```

Confirmed with Dapr 1.10.4 (`git checkout v1.10.4`).

## Release Note

RELEASE NOTE: **FIX** Updating a reminder silently fails if the sidecar is not an actor host

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
Fixes #6150

# Description

Although endpoints to modify actor reminders are meant to be invoked by running actors only, there's currently nothing preventing them from being invoked by anyone. This causes interesting issues such as #6150, where operations on reminders can fail if they are being executed by a sidecar that doesn't host actors of the given type.

This implements the fix discussed on the issue (https://github.com/dapr/dapr/issues/6150#issuecomment-1489665935), disabling operations on actor reminders if the sidecar isn't hosting actors of the given type.

This should be considered a **breaking change**, even if (hopefully) of small impact. It prevents behaviors that were never intended to be used (but may have been used by someone, per Hyrum's law) and that were buggy and not always working as intended.

## Release Note

RELEASE NOTE: **FIXED** Disallow operations on reminders for actor types that are not currently being hosted

### CODE REVIEW

File: pkg/actors/actors.go [hunk 1]
```diff
@@ -1861,6 +1874,10 @@ func (a *actorsRuntime) storeReminder(ctx context.Context, store transactionalSt
 }
 
 func (a *actorsRuntime) GetReminder(ctx context.Context, req *GetReminderRequest) (*reminders.Reminder, error) {
+	if !a.config.HostedActorTypes.IsActorTypeHosted(req.ActorType) {
+		return nil, ErrReminderOpActorNotHosted
```
Comment (daixiang0) [inline pkg/actors/actors.go:1878]: Could add a test to cover this case?

Comment (ItalyPaleAle) [reply pkg/actors/actors.go:1878]: There are unit tests in pkg/http that cover this scenario, as well as E2E tests.

Additionally the unit tests in pkg/actors have been updated to make the actor type hosted, so the positive case is included.

I think that should be enough?

**Review by daixiang0 — APPROVED** [review]
LGTM, wait for CI happy

**Comment by ItalyPaleAle** [comment]
Per conversation with @yaron2 over Discord this change should not be considered a major breaking change, although it has a chance of breaking a small number of users. However:

- It seems that SDKs (or at least the ones I've looked at) don't allow creating reminders for actors that aren't hosted anyways, so this _may_ only impact users that directly invoke the Actor APIs - arguably a small minority
- Users who did end up invoking this method to create reminders for actors that aren't hosted would have run into a number of issues. Even when the operation succeeded, it could have caused issues downstreams.

I have added the "breaking change" label so we remember to document this change clearly. However, maintainers don't seem concerned with it particulary.

**Pull request merged**